dtls.c

```diff
@@ -3628,10 +3633,11 @@ handle_handshake_msg(dtls_context_t *ctx, dtls_peer_t *peer, uint8 *data, size_t
 #ifdef DTLS_ECC
   case DTLS_HT_CERTIFICATE_REQUEST:
 
-    if (state != DTLS_STATE_WAIT_SERVERHELLODONE) {
+    if (!is_tls_ecdhe_ecdsa_with_aes_128_ccm_8(peer->handshake_params->cipher) ||
```

Comment:
Hm, the newly introduced first check seems to be unrelated to this PR. This might warrant at least a separate commit.

Reply:
If you prefer it, sure.